File: mpp-proiect-repository-cristinahendre/lab6(web)/CSharp/Lab5/Domain/CazCaritabil.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Cerinta1.Domain
{
    [Serializable]
    public class CazCaritabil : Entity<int>
    {
        string nume;
        int suma_donata;

        public CazCaritabil(string nume, int suma_donata)
        {
            this.nume = nume;
            this.suma_donata = suma_donata;
        }

        public virtual  int Suma_donata { get => suma_donata; set => suma_donata = value; }
        public virtual string Nume { get => nume; set => nume = value; }

        public override bool Equals(object obj)
        {
            return obj is CazCaritabil caritabil &&
                   nume == caritabil.nume &&
                   suma_donata == caritabil.suma_donata;
        }

        public override int GetHashCode()
        {
            int hashCode = -1494288119;
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(nume);
            hashCode = hashCode * -1521134295 + suma_donata.GetHashCode();
            return hashCode;
        }

        public override string ToString()
        {
            return "Id = " + this.Id + " ,nume = " + Nume + " " + " ,suma donata = " + Suma_donata;
        }
    }
}

File: mpp-proiect-repository-cristinahendre/lab12/lab6/modul/DomainModule/src/main/java/Domain/CazCaritabil.java
package Domain;

import java.io.Serializable;
import java.util.Objects;

public class CazCaritabil extends Entity<Integer> implements Serializable {
    private String nume;
    private int suma_donata;

    public CazCaritabil(){}
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof CazCaritabil)) return false;
        CazCaritabil that = (CazCaritabil) o;
        return getSuma_donata() == that.getSuma_donata() && getNume().equals(that.getNume());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getNume(), getSuma_donata());
    }

    public String getNume() {
        return nume;
    }

    @Override
    public String toString() {
        return "CazCaritabil{" +
                "id= "+ this.getId()+" "+
                "nume='" + nume + '\'' +
                ", suma_donata=" + suma_donata +
                '}';
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public int getSuma_donata() {
        return suma_donata;
    }

    public void setSuma_donata(int suma_donata) {
        this.suma_donata = suma_donata;
    }

    public CazCaritabil(String nume) {
        this.nume = nume;
    }
}
